### PR TITLE
[bugfix] Fix JS metric

### PR DESCRIPTION
**Pull request by milah**
*Issue #, if available:* P44014016

*Description of changes:*
According to our developer [guide](https://docs.aws.amazon.com/sagemaker/latest/dg/clarify-data-bias-metric-jensen-shannon-divergence.html) for JS metric (which is aligned with the formula on wiki Jensen-Shannon divergence)

The formula for the Jensen-Shannon divergence is as follows:

   JS = ½*[KL(Pa || P) + KL(Pd || P)]

Where P = ½( Pa + Pd ), the average label distribution across facets a and d.

Instead we currently use the label distribution in the dataset. This PR fixes this bug reported by Xiaoguang.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

Tested: used in the Container as package.

### CODE REVIEW

File: src/smclarify/bias/metrics/posttraining.py [hunk 1]
```diff
@@ -317,15 +318,15 @@ def TE(
     return te
 
 
-def FlipSet_pos(dataset: np.array, labels: np.array, predicted_labels: np.array) -> np.array:
+def FlipSet_pos(dataset: List[Any], labels: List[Any], predicted_labels: np.ndarray) -> np.ndarray:
```
Comment (AnkitSiva) [inline src/smclarify/bias/metrics/posttraining.py:321]: Could you be more specific than `List[Any]`?

File: tests/unit/bias/metrics/test_metrics.py [hunk 2]
```diff
@@ -291,7 +292,18 @@ def test_JS():
     res = JS(positive_label_index, sensitive_facet_index)
     assert res == approx(0.06465997)
 
-    return
+    # Example added from P44014016.
```
Comment (pinaraws) [inline tests/unit/bias/metrics/test_metrics.py:295]: please remove this comment (internal issue number)

**Review by pinaraws — CHANGES_REQUESTED** [review]

File: src/smclarify/bias/metrics/pretraining.py [hunk 3]
```diff
@@ -108,11 +108,11 @@ def JS(label: pd.Series, sensitive_facet_index: pd.Series) -> float:
     require(sensitive_facet_index.dtype == bool, "sensitive_facet_index must be of type bool")
     xs_a = label[~sensitive_facet_index]
     xs_d = label[sensitive_facet_index]
-    (Pa, Pd, P) = pdfs_aligned_nonzero(xs_a, xs_d, label)
+    (Pa, Pd, _) = pdfs_aligned_nonzero(xs_a, xs_d, label)
+    P = 1 / 2 * (Pa + Pd)
     if len(Pa) == 0 or len(Pd) == 0 or len(P) == 0:
```
Comment (xgchena) [inline src/smclarify/bias/metrics/pretraining.py:113]: We don't need to pass `label` to the function anymore because `xs_a` and `xa_d` together have all values of `label`. Code like below should work fine,

```
..
(Pa, Pd) = pdfs_aligned_nonzero(xs_a, xs_d)
P = 0.5 * (Pa + Pd)
if len(Pa) == 0 or len(Pd) == 0:
...

File: src/smclarify/bias/metrics/pretraining.py [hunk 4]
```diff
@@ -108,11 +108,11 @@ def JS(label: pd.Series, sensitive_facet_index: pd.Series) -> float:
     require(sensitive_facet_index.dtype == bool, "sensitive_facet_index must be of type bool")
     xs_a = label[~sensitive_facet_index]
     xs_d = label[sensitive_facet_index]
-    (Pa, Pd, P) = pdfs_aligned_nonzero(xs_a, xs_d, label)
+    (Pa, Pd, _) = pdfs_aligned_nonzero(xs_a, xs_d, label)
```
[the remark this answers is not in the document]
Comment (milah) [reply src/smclarify/bias/metrics/pretraining.py:111]: done

File: src/smclarify/bias/metrics/posttraining.py [hunk 5]
```diff
@@ -13,6 +13,7 @@
 import numpy as np
 from sklearn.neighbors import KNeighborsClassifier
 from smclarify.bias.metrics.constants import INFINITY, FT_DEFAULT_NEIGHBOR, FT_MIN_NEIGHBOR, FT_SAMPLES_COUNT_THRESHOLD
+from typing import List
```
Comment (pinaraws) [inline src/smclarify/bias/metrics/posttraining.py:16]: this change is not needed, you can remove it.

**Review by pinaraws — APPROVED** [review]
One nit  pick, rest looks good to me, thanks!

File: src/smclarify/bias/metrics/posttraining.py (hunk 5 above)
Comment (milah) [reply src/smclarify/bias/metrics/posttraining.py:16]: good catch!

**Review by xgchena — APPROVED** [review]

**Review by keerthanvasist — APPROVED** [review]
